code/modules/admin/admin_ranks.dm

```diff
@@ -199,6 +199,33 @@ GLOBAL_PROTECT(protected_ranks)
 	testing(msg)
 	#endif
 
+/// Converts a rank name (such as "Coder+Moth") into a list of /datum/admin_rank
+/proc/ranks_from_rank_name(rank_name)
+	var/list/rank_names = splittext(rank_name, "+")
+	var/list/ranks = list()
+
+	for (var/datum/admin_rank/rank as anything in GLOB.admin_ranks)
+		if (rank.name in rank_names)
+			rank_names -= rank.name
+			ranks += rank
+
+			if (rank_names.len == 0)
```

Comment:
why not !length(rank_names)?

Reply:
I don't like treating non-bool 0 as falsy unless it happens to help in some other way (such as also testing empty strings/false at the same time).

Reply:
^ cringe